templates/robot_description/robot_moveit.launch.py

```diff
@@ -0,0 +1,188 @@
+from launch.conditions import IfCondition, UnlessCondition
+from launch.substitutions import Command, FindExecutable, LaunchConfiguration, PathJoinSubstitution
+from launch_ros.actions import Node
+from launch_ros.substitutions import FindPackageShare
+
+
+def load_yaml(package_name, file_path):
+    package_path = get_package_share_directory(package_name)
+    absolute_file_path = os.path.join(package_path, file_path)
+
+    try:
+        with open(absolute_file_path) as file:
+            return yaml.safe_load(file)
+    except OSError:  # parent of IOError, OSError *and* WindowsError where available
+        return None
+
+
+def launch_setup(context, *args, **kwargs):
+    use_sim_time = LaunchConfiguration("use_sim_time")
+    launch_rviz = LaunchConfiguration("launch_rviz")
+    only_rviz = LaunchConfiguration("only_rviz")
+
+    # Get SRDF via xacro
+    robot_srdf_content = Command(
+        [
+            PathJoinSubstitution([FindExecutable(name="xacro")]),
+            " ",
+            PathJoinSubstitution(
+                [FindPackageShare("$PKG_NAME$"), "srdf", "$ROBOT_NAME$.srdf.xacro"]
+            ),
+        ]
+    )
+    robot_description_semantic = {
+        "robot_description_semantic": robot_srdf_content.perform(context)
+    }
+
+    moveit_config_package_name = "$ROBOT_NAME$_moveit_config"
+
+    controllers_yaml = load_yaml(
+        moveit_config_package_name, "config/$ROBOT_NAME$/controllers.yaml"
+    )
+    ompl_planning_yaml = load_yaml(
+        moveit_config_package_name, "config/$ROBOT_NAME$/ompl_planning.yaml"
+    )
+    kinematics_yaml = load_yaml(moveit_config_package_name, "config/$ROBOT_NAME$/kinematics.yaml")
+    robot_description_kinematics = {"robot_description_kinematics": kinematics_yaml}
+    limits_yaml = load_yaml(moveit_config_package_name, "config/$ROBOT_NAME$/joint_limits.yaml")
+    robot_description_planning = {"robot_description_planning": limits_yaml}
+
+    # Planning Configuration
+    ompl_planning_pipeline_config = {
+        "default_planning_pipeline": "ompl",
+        "planning_pipelines": ["ompl"],
+        "ompl": {
+            "planning_plugin": "ompl_interface/OMPLPlanner",
+            "request_adapters": """default_planner_request_adapters/AddTimeOptimalParameterization default_planner_request_adapters/FixWorkspaceBounds default_planner_request_adapters/FixStartStateBounds default_planner_request_adapters/FixStartStateCollision default_planner_request_adapters/FixStartStatePathConstraints""",
+            "start_state_max_bounds_error": 0.1,
+        },
+    }
+    ompl_planning_pipeline_config["ompl"].update(ompl_planning_yaml)
+
+    # Moveit controllers Configuration
+    moveit_controllers = {
+        "moveit_simple_controller_manager": controllers_yaml,
+        "moveit_controller_manager": "moveit_simple_controller_manager/MoveItSimpleControllerManager",
+    }
+
+    # Trajectory Execution Configuration
+    trajectory_execution = {
```

Comment:
If we end up creating a `moveit_config` package template, it would be nice to put these miscelaneous settings into one .yaml file, so that we don't bloat the launch file with dictionaries.
For example:
- `moveit_manage_controllers`
- `trajectory_execution.*`

Now when I think about it, we are carrying this .yaml structure from way back in `UniversalRobots`, from where we always copied the moveit configuration for our applications. Now that we know what we usually set and where, I bet we can come up with a sensible structure for `.yaml` config files.


#### Edit:
I see there is `planning_pipelines.yaml` and `moveit_execution_and_control.yaml` commented out below, `Line 127`-onwards, is this the same idea i described above?

Reply:
Yes, exactly! @riyan-jose you have been working on putting all MoveIt parameters to `*.yaml` files. Can you help @Nibanovic create tempaltes here.